drivers/storage/portworx/component/telemetry.go

```diff
@@ -59,16 +83,360 @@ func (t *telemetry) Delete(cluster *corev1.StorageCluster) error {
 	if err := k8sutil.DeleteConfigMap(t.k8sClient, TelemetryConfigMapName, cluster.Namespace, *ownerRef); err != nil {
 		return err
 	}
+	t.MarkDeleted()
 	return nil
 }
 
-func (t *telemetry) MarkDeleted() {}
+func (t *telemetry) MarkDeleted() {
+	t.isCollectorDeploymentCreated = false
+}
 
 // RegisterTelemetryComponent registers the telemetry  component
 func RegisterTelemetryComponent() {
 	Register(TelemetryComponentName, &telemetry{})
 }
 
+func (t *telemetry) deployMetricsCollector(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	if err := t.createProxyConfigMap(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorConfigMap(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorRole(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorRoleBinding(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorDeployment(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *telemetry) createCollectorDeployment(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	if t.isCollectorDeploymentCreated {
+		return nil
+	}
+
+	replicas := int32(1)
+	labels := map[string]string{
+		"role": "realtime-metrics-collector",
+	}
+	runAsUser := int64(1111)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            CollectorDeploymentName,
+			Namespace:       cluster.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{MatchLabels: labels},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{Labels: labels},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "collector",
+							Image: manifest.DefaultCollectorImage,
```

Comment:
Using images from the manifest is fine for this PR, but we need to change this to use `cluster.Status.DesiredImages` to populate the component images. Check other components for that. We can do that as part of [OPERATOR-491](https://portworx.atlassian.net/browse/OPERATOR-491)